src/main/java/io/github/klsmith/budgettracker/money/budget/BudgetController.java

```diff
@@ -0,0 +1,109 @@
+import spark.Service;
+
+public class BudgetController {
+
+    private static final Logger logger = LoggerFactory.getLogger(BudgetController.class);
+
+    private static final String ID_PARAM = ":idParam";
+    private static final String RESPONSE_TYPE_APPLICATION_JSON = "application/json";
+
+    private final BudgetService service;
+
+    public BudgetController(BudgetService service) {
+        this.service = Objects.requireNonNull(service,
+                String.format("Cannot have a null %s.",
+                        BudgetService.class.getSimpleName()));
+    }
+
+    public void route(Service spark) {
+        spark.post("/api/money/budget", this::postBudget);
+        spark.get("/api/money/budget/" + ID_PARAM, this::getBudgetById);
+        spark.put("/api/money/budget/" + ID_PARAM, this::putBudgetById);
+        spark.delete("/api/money/budget/" + ID_PARAM, this::deleteBudgetById);
+    }
+
+    public String postBudget(Request request, Response response) {
+        response.type(RESPONSE_TYPE_APPLICATION_JSON);
+        return parseJson(request.body())
+                .map(service::create)
+                .map(this::toJson)
+                .orElseThrow(RuntimeException::new);
+    }
+
+    public String getBudgetById(Request request, Response response) {
+        response.type(RESPONSE_TYPE_APPLICATION_JSON);
+        final long id = Long.parseLong(request.params(ID_PARAM));
+        return service.find(id)
+                .map(this::toJson)
+                .orElseGet(() -> {
+                    response.status(404);
+                    return String.format("{ \"message\"=\"Cannot find %s with id=%s\" }",
+                            Budget.class.getSimpleName(), Long.valueOf(id));
+                });
+    }
+
+    public String putBudgetById(Request request, Response response) {
+        response.type(RESPONSE_TYPE_APPLICATION_JSON);
+        final long id = Long.parseLong(request.params(ID_PARAM));
+        return parseJson(request.body())
+                .flatMap(budget -> service.update(id, budget))
+                .map(this::toJson)
+                .orElseGet(() -> {
+                    response.status(404);
+                    return String.format("{ \"message\"=\"Cannot find %s with id=%s\" }",
+                            Budget.class.getSimpleName(), Long.valueOf(id));
+                });
+    }
+
+    public String deleteBudgetById(Request request, Response response) {
+        response.type(RESPONSE_TYPE_APPLICATION_JSON);
+        final long id = Long.parseLong(request.params(ID_PARAM));
+        service.delete(id);
+        return String.format("{ \"message\"=\"Successfully deleted %s with id=%s\" }",
+                Budget.class.getSimpleName(), Long.valueOf(id));
+    }
+
+    private String toJson(Budget budget) {
+        return new Gson().toJson(budget);
+    }
+
+    private Optional<Budget> parseJson(String json) {
+        try {
+            final JsonObject root = new JsonParser()
```

Comment:
MAYBE refactor "root" to "jsonObject". It feels slightly more descriptive. (Obviously this is very minute and open to argument)

Reply:
I will be honest, naming something the same as the class name does not make it more descriptive; however, that doesn't mean "root" is better.

My reasoning was that I wanted to be aware while coding that it was the root-level object that contains all of the data. I will reread the code again and consider changing it.

Reply:
I decided I do actually prefer root, to me it tells me exactly what I need to know, within the context of the method.